taichi/backends/opengl/opengl_api.cpp

```diff
@@ -222,7 +222,13 @@ bool initialize_opengl(bool use_gles, bool error_tolerance) {
 
 void CompiledTaichiKernel::init_args(Kernel *kernel) {
   arg_count = kernel->args.size();
-  ret_count = kernel->rets.size();
+  ret_count = 0;
+  for (auto &ret : kernel->rets) {
+    if (auto tensor_type = ret.dt->cast<TensorType>())
+      ret_count += tensor_type->get_num_elements();
+    else
+      ret_count += 1;
+  }
```

Comment:
Is `ret_count` used anywhere?

Reply:
Yes, it was used to compute return buffer size. Although it is now unuseful in the current version of taichi.